app/models/document.rb

```diff
@@ -138,6 +138,14 @@ def humanized_document_type
     document_type.underscore.tr("_", " ")
   end
 
+  def update_edition_references!
+    latest = editions.reverse_order
+    update!(
+      latest_edition_id: latest.pick(:id),
```

Comment:
I think we need to exclude editions which are deleted

```
latest = editions.where.not(state:  "deleted:).reverse_order
```

Reply:
I think this is already covered due to `Edition` having that tricksy default scope (😄😭)

https://github.com/alphagov/whitehall/blob/811ff58d063e7306fbf6d4a25f00f134c61a4c37/app/models/edition/workflow.rb#L21

However, I could add a deleted edition to the unit test for this method. That'll at least avoid a regression if ever we come to change the default scope.

What do you think?

Reply:
oh right yea good point. I always forget about the scoped find. All good as is then 👍

Reply:
The default scope is the sort of thing that now seems like a necessary evil, because the whole app has been built around the assumption that it can never accidentally find a "deleted" Edition.

This does seem like an obvious 'new' case of that assumption being used though.

I might try throwing in a deleted Edition to this method's unit test – just to explicitly declare that assumption & avoid an accidental regression.